### PR TITLE
Added BlendFactor property

### PR DESCRIPTION
**Pull request by Shqrdx**
remaked https://github.com/mono/MonoGame/pull/4141

### CODE REVIEW

File: MonoGame.Framework/Graphics/States/BlendState.DirectX.cs [hunk 1]
```diff
@@ -38,19 +38,19 @@ internal void PlatformApplyState(GraphicsDevice device)
 
             Debug.Assert(GraphicsDevice == device, "The state was created for a different device!");
 
-			// NOTE: We make the assumption here that the caller has
-			// locked the d3dContext for us to use.
+            // NOTE: We make the assumption here that the caller has
+            // locked the d3dContext for us to use.
```
Comment (Shqrdx) [inline MonoGame.Framework/Graphics/States/BlendState.DirectX.cs]: Sorry about formatting, looks like resharper did it and I didnt notice..

File: MonoGame.Framework/Graphics/GraphicsDevice.cs [hunk 2]
```diff
@@ -324,6 +325,11 @@ public RasterizerState RasterizerState
             }
         }
 
+        /// <summary>
+        /// Blend factor for alpha blending.
+        /// </summary>
+        public Color BlendFactor { get { return _blendFactor; } set { _blendFactor = value; } }
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/GraphicsDevice.cs]: This won't work.

You can change the blend factor here yet the blend state won't reapply until `_blendStateDirty` is set to true.

Comment (Shqrdx) [inline MonoGame.Framework/Graphics/GraphicsDevice.cs]: Yeah, true..

**Comment by Shqrdx** [comment]
@tomspilman Fixed

**Comment by Shqrdx** [comment]
@tomspilman @KonajuGames It works, merge ?

**Pull request closed without merging**